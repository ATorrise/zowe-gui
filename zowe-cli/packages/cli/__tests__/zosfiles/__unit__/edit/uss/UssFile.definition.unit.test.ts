/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import i18nEnglish from "../../../../../src/zosfiles/-strings-/en";
import { USSFileDefinition } from "../../../../../src/zosfiles/edit/uss/USSFile.definition";

describe("DatasetDefinition", () => {
    it("should use the correct object for strings", () => {
        const strings = i18nEnglish.EDIT.ACTIONS.USS_FILE;

        expect(USSFileDefinition.description).toBe(strings.DESCRIPTION);
        expect(USSFileDefinition.children).toBeUndefined();
        expect(USSFileDefinition.options).toMatchSnapshot();
        expect(USSFileDefinition.examples).toMatchSnapshot();
    });
});
