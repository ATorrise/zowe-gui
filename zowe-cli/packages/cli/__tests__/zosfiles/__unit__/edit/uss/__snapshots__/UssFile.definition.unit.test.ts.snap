// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetDefinition should use the correct object for strings 1`] = `
Array [
  Object {
    "aliases": Array [
      "ed",
    ],
    "description": "Editor that overrides the default editor for this file type. Set the option to the editor's executable file location or the program's name: ie \\"--editor notepad\\"",
    "name": "editor",
    "required": false,
    "type": "string",
  },
  Object {
    "aliases": Array [
      "b",
    ],
    "description": "Transfer the file content in binary mode (no EBCDIC to ASCII conversion).",
    "name": "binary",
    "type": "boolean",
  },
  Object {
    "aliases": Array [
      "ec",
    ],
    "conflictsWith": Array [
      "binary",
    ],
    "description": "Transfer the file content with encoding mode, which means that data conversion is performed using the file encoding specified.",
    "name": "encoding",
    "type": "string",
  },
]
`;

exports[`DatasetDefinition should use the correct object for strings 2`] = `
Array [
  Object {
    "description": "Edit the contents of the USS file \\"/a/ibmuser/my_text.txt\\" in notepad",
    "options": "/a/ibmuser/my_text.txt --editor notepad",
  },
  Object {
    "description": "Edit the contents of the USS file \\"/a/ibmuser/my_text.txt\\" in notepad",
    "options": "/a/ibmuser/my_text.txt --editor C:\\\\Windows\\\\System32\\\\Notepad.exe",
  },
  Object {
    "description": "Edit the contents of the USS file \\"/a/ibmuser/my_jcl.jcl\\" with binary flag set",
    "options": "/a/ibmuser/my_text.txt --binary",
  },
]
`;
